packages/modules/imodel-browser/src/containers/iModelGrid/useIModelData.ts

```diff
@@ -111,7 +122,10 @@ export const useIModelData = ({
       })
       .then((result: { iModels: IModelFull[] }) => {
         setStatus(DataStatus.Complete);
-        if (result.iModels.length !== PAGE_SIZE) {
+        if (
+          result.iModels.length !== PAGE_SIZE ||
+          result.iModels.length === pageSize
```

Comment:
We can't call this `pageSize` if this is also an overall limit, I think it would make more sense to call that `top` or `maxCount`. (Because if someone ever wants to add a real "paging", then they will be able to use the `skip` as well and only get the ones they really want, the `page` here is an internal concept as this is infinite scroll, not a paging system. (So it will be confusing API wise to mention a "page size", when we dont have any pages displayed...)

Also, in which usecase does this make sense that a user only wants the top 10 iModels sorted by name ?

Reply:
Updated.

In studio we have a useCase where we have a AvailableIModels section on HomePage and we only want to display the 7 iModels and we do have a button viewAll which will basically shows all iModels in a fronstage.